data_analysis/pii/utils/evaluation.py

```diff
@@ -0,0 +1,103 @@
+import json
+
+TAGS = ['EMAIL', 'IP_ADDRESS', 'KEY']
+
+
+def load_json(sample):
+    try:
+        return json.loads(sample)
+    except ValueError:
+        return []
+
+
+def overlapped(a, b, alpha=0.8, beta=0.8):
+    """Returns True if the intervals a and b overlap for more than 80% of their lengths"""
+    size_overlap = max(0, min(a[1], b[1]) - max(a[0], b[0]))
+    ref_overlap = size_overlap / (b[1] - b[0])
+    pred_overlap = size_overlap / (a[1] - a[0])
+    return (ref_overlap > alpha and pred_overlap > beta)
+
+
+def compare_intervals(references, predictions, alpha=0.8, beta=0.8):
+    """Compare two lists of intervals and return the number 
+    of true positives, false positives and false negatives.
+    >>> compare_intervals([(0, 7), (10, 20)], [(1,8), (99, 119)], 0, 0)[0]
+    {'TP': 1, 'FN': 1, 'FP': 1}
+    """
+    ref_intervals = sorted(references, key=lambda x: x[0])
+    pred_intervals = sorted(predictions, key=lambda x: x[0])
+    scores = {"TP": 0, "FN": 0, "FP": 0}
+    detected_secrets = []
+    for interval in pred_intervals:
```

Comment:
The implementation here is quadratic. But it should be ok as there should be a limited number of pii per file